pvanalytics/quality/irradiance.py

```diff
@@ -471,3 +471,174 @@ def daily_insolation_limits(irrad, clearsky, daily_min=0.4, daily_max=1.25):
+        datetime index as ghi, dni, and zenith series, when available.
+    dni : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dhi, and zenith series, when available.
+    dni_clear : Series, optional
+        Pandas series of clearsky dni data. Must have the same datetime index
+        as ghi, dhi, dni, and zenith series, when available. See
+        :py:func:`dni` for details.
+    Returns
+    -------
+    component_sum_df : Dataframe
+        Pandas series of 'ghi', 'dhi', and 'dni' columns with datetime index
+    """
+    if ghi is not None and dhi is not None and dni is None:
+        dni = pvlib.irradiance.dni(ghi, dhi, solar_zenith,
+                                   clearsky_dni=dni_clear,
+                                   clearsky_tolerance=1.1)
+    elif dni is not None and dhi is not None and ghi is None:
+        ghi = (dhi + dni * cosd(solar_zenith))
+    elif dni is not None and ghi is not None and dhi is None:
+        dhi = (ghi - dni * cosd(solar_zenith))
+    else:
+        raise ValueError(
+            "Please check that exactly one of ghi, dhi and dni parameters "
+            "is set to None"
+        )
+    # Merge the outputs into a master dataframe containing 'ghi', 'dhi',
+    # and 'dni' columns
+    component_sum_df = pd.DataFrame({'ghi': ghi,
+                                     'dhi': dhi,
+                                     'dni': dni})
+    return component_sum_df
+
+
+def calculate_component_sum_series(solar_zenith,
+                                   ghi=None,
+                                   dhi=None,
+                                   dni=None,
+                                   dni_clear=None,
+                                   zenith_limit=90,
+                                   fill_value=np.nan,
+                                   fill_nighttime=None):
+    '''
+    Use the component sum equations to calculate the missing series, using
+    the other available time series. One of the three parameters (ghi, dhi,
+    dni) is passed as None, and the two series are used to
+    calculate the missing series. After calculation, the series is
+    run through a nighttime routine, where nighttime values are set based on
+    the fill_value and fill_nighttime parameters.
+
+    The "component sum" or "closure" equation relates the three
+    primary irradiance components as follows:
+    .. math::
+       GHI = DHI + DNI * cos(theta_z)
+
+    Parameters
+    ----------
+    solar_zenith : Series
+        Zenith angles in decimal degrees, with datetime index.
+        Angles must be >=0 and <=180. Must have the same datetime index
+        as ghi, dhi, and dni series, when available.
+    ghi : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as dni, dhi, and zenith series, when available.
```

Comment:
```suggestion
        Pandas series of GHI data, with datetime index. Must have the same
        datetime index as dni, dhi, and solar_zenith series, when available.
```

Similar edits needed for dhi, dni, and dni_clear.